### PR TITLE
exe-config: Fix underflow in ghcjs config lookup (Fixes #465)

### LINKED ISSUE #465: Exe app/ghcjs causes mem leak/crash with prerender
This works fine in `ob run`, but with `nix-build -A exe` app the browser only shows "test" and then it crashes due to memory leak. Could be some issue with hydration.
```
  , _frontend_body = do
      text "test"
      void $ prerender (pure ()) $ do
        text "Welcome to Obelisk!"
        el "p" $ text $ T.pack commonStuff
        elAttr "img" ("src" =: static @"obelisk.jpg") blank
```

### PR DESCRIPTION
**Pull request by tomsmalley**
Fixes #465

### CODE REVIEW

File: lib/executable-config/lookup/src-ghcjs/Obelisk/ExecutableConfig/Lookup.hs [hunk 1]
```diff
@@ -35,7 +35,7 @@ getConfigs = do
   where
     collToList es = do
       len <- getLength es
-      list <- traverse (item es) [0..len-1]
+      list <- traverse (item es) $ if len == 0 then [] else [0..len-1]
```
Comment (3noch) [inline lib/executable-config/lookup/src-ghcjs/Obelisk/ExecutableConfig/Lookup.hs]: And this, folks, is why subtraction is not a total function.

Comment (3noch) [reply lib/executable-config/lookup/src-ghcjs/Obelisk/ExecutableConfig/Lookup.hs]: Super nitpick idea: `[0..max 0 (len-1)]`

Comment (tomsmalley) [reply lib/executable-config/lookup/src-ghcjs/Obelisk/ExecutableConfig/Lookup.hs]: That actually behaves a little differently when `len = 0`

Comment (3noch) [reply lib/executable-config/lookup/src-ghcjs/Obelisk/ExecutableConfig/Lookup.hs]: Aha right I forgot it was inclusive.

Comment (3noch) [reply lib/executable-config/lookup/src-ghcjs/Obelisk/ExecutableConfig/Lookup.hs]: Haha.... wow. It's hard to deal with under/over flow! If subtraction were `Word -> Word -> Maybe Word` then `max 0 (len-1)` wouldn't type check either. Super tricky.

Comment (3noch) [reply lib/executable-config/lookup/src-ghcjs/Obelisk/ExecutableConfig/Lookup.hs]: I usually just avoid unsigned types like the plague for this reason. I'm just curious why we aren't.

**Comment by alexfmpe** [comment]
Yikes. Is there no way to traverse the collection directly without messing with indexes?
Failing that, maybe we should extract/upstream this fix into an util?

**Comment by 3noch** [comment]
Should we make this impl match what seems to be the same bug fixed elsewhere?

https://github.com/obsidiansystems/obelisk/blob/86a9584c6d7648bd5324ef57d62421fed1bf1978/lib/frontend/src/Obelisk/Frontend.hs#L113

**Comment by 3noch** [comment]
Your fix avoids `fromIntegral` so maybe you should actually port your fix over to that line as well. Also a comment would be great.

**Comment by 3noch** [comment]
Is there a version of `fromIntegral` that throws/`Maybe`s result so that we could "safely" cast to convert to signed type? I would much prefer simply getting rid of the unsignedness altogether.

**Comment by tomsmalley** [comment]
Interesting, didn't even notice that one. Looks like `collToList` itself should be factored out

**Comment by 3noch** [comment]
@tomsmalley This bug is bad enough that finding the perfect solution to abstracting out this function is definitely not worth it if it keeps us from getting rid of it. Let's just make the two implementations the same for now with a big TODO on both of them. I'll approve this and hopefully we can get it merged. I'll try to make the change if you don't beat me to it.

**Pull request merged**